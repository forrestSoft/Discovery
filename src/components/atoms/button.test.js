import React from 'react'
import {shallow, mount} from 'enzyme'
import shallowToJson from 'enzyme-to-json'

import Button from './button'

it('render matches snapshot', () => {
  const button = shallowToJson(shallow(
  	<Button 
  		text={'I am a button'}
  		disabled={false}
  	/>
  )) 
  expect(button).toMatchSnapshot();
});

it('render disabled matches snapshot', () => {
  const button = shallowToJson(shallow(
  	<Button 
  		text={'I am a button'}
  		disabled={true}
  	/>
  )) 
  expect(button).toMatchSnapshot();
});

it('calls passed onClick on "onClick"', () => {
	const mock = jest.fn().mockReturnValue('onClick')
	const button = mount(
  	<Button 
  		text={'I am a button'}
  		disabled={false}
  		handleClick={mock}
  	/>
  )

	button.find('button').simulate('click')
	expect(mock).toBeCalled()
})

it('doesnt call passed onClick on "onClick" when disabled', () => {
  const mock = jest.fn().mockReturnValue('onClick')
  const button = mount(
    <Button 
      text={'I am a button'}
      disabled={true}
      handleClick={mock}
    />
  )
  
  button.find('button').simulate('click')
  expect(mock).not.toBeCalled()
})