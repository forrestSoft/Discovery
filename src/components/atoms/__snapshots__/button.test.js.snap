// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render disabled matches snapshot 1`] = `
<button
  disabled={true}
>
  I am a button
</button>
`;

exports[`render matches snapshot 1`] = `
<button
  disabled={false}
>
  I am a button
</button>
`;
